dipy/direction/pmf.pyx

```diff
@@ -14,15 +15,21 @@ from dipy.core.interpolation cimport trilinear_interpolate4d_c
 cdef class PmfGen:
 
     def __init__(self,
-                 double[:, :, :, :] data):
-        self.data = np.asarray(data,  dtype=float)
+                 double[:, :, :, :] data,
+                 object sphere=default_sphere):
```

Comment:
After a second thought, I agree with you, let's remove the `default_sphere`. It should be mandatory and not optional.

Reply:
Ok, (sorry I delete the previous comment as I making additional tests.). Ok, `SHCoeffPmfGen` can use any spheres as input, but the user must be careful with `SimplePmfGen` and setting a default sphere might lead to erroneous results hard to debug.